flash/core/data/data_source.py

```diff
@@ -697,3 +709,218 @@ def _get_classes(self, data):
+                # extracting data types from tasks
+                for key in task.get("data"):
+                    self.data_types.add(key)
+                # Adding ground_truth annotation to separate dataset
+                result = annotation["result"]
+                for res in result:
+                    t = res["type"]
+                    for label in res["value"][t]:
+                        # check if labeling result is a list of labels
+                        if isinstance(label, list) and not self.multi_label:
+                            for sublabel in label:
+                                self.classes.add(sublabel)
+                                temp = {}
+                                temp["file_upload"] = task.get("file_upload")
+                                temp["data"] = task.get("data")
+                                temp["label"] = sublabel
+                                temp["result"] = res.get("value")
+                                if annotation["ground_truth"]:
+                                    self.test_results.append(temp)
+                                elif not annotation["ground_truth"]:
+                                    self.results.append(temp)
+                        else:
+                            if isinstance(label, list):
+                                for item in label:
+                                    self.classes.add(item)
+                            else:
+                                self.classes.add(label)
+                            temp = {}
+                            temp["file_upload"] = task.get("file_upload")
+                            temp["data"] = task.get("data")
+                            temp["label"] = label
+                            temp["result"] = res.get("value")
+                            if annotation["ground_truth"]:
+                                self.test_results.append(temp)
+                            elif not annotation["ground_truth"]:
+                                self.results.append(temp)
+        self.num_classes = len(self.classes)
+        # splitting result to train and val sets
+        import random
+
+        random.shuffle(self.results)
+        data_length = len(self.results)
+        prop = data_length - int(data_length * self.split)
+        self.val_results = self.results[:prop]
+        self.results = self.results[prop:]
+
+    def load_sample(self, sample: Mapping[str, Any] = None, dataset: Optional[Any] = None) -> Any:
+        """Load 1 sample from dataset."""
+        # all other data types
+        input_data = deepcopy(sample)
+        try:
+            del input_data["label"]
+        except KeyError:
+            # no label in input data
+            pass
+        result = {
+            DefaultDataKeys.INPUT: input_data,
+            DefaultDataKeys.TARGET: self._get_labels_from_sample(sample["label"]),
+        }
+        return result
+
+    def generate_dataset(
```

Comment:
I don't think this would work if a user provide only a test or predict dataset ?

I would prefer the from_labelstudio to provide a way to split dataset before hand.

The API should looks like this.

```py
# 1. Load export data
datamodule = VideoClassificationData.from_labelstudio(
    train_export_json="data/train_export_json.json",
    val_export_json="data/val_export_json.json",
    test_export_json="data/test_project.json",
    predict_export_json="data/predict_project.json",
    data_folder="data/upload/",    # lowest priority, will be overriden by {state}_data_folder
    train_data_folder="data/train_upload/",
    val_data_folder="data/val_upload/",
    test_data_folder="data/test_upload/",
    predict_data_folder="data/predit_upload/",
    val_split=0.2,
    clip_sampler="uniform",
    clip_duration=1,
    decode_audio=False,
)
```

Reply:
There should be priority of `train_export_json` + `train_data_folder`  over `train_export_json` + `data_folder`.

And `train_export_json`  can be used to generate `val_export_json` internally using `val_split` is `val_export_json` wasn’t provided.

Reply:
Fixed